lib/gears/timer.lua

```diff
@@ -1,6 +1,47 @@
 ---------------------------------------------------------------------------
 --- Timer objects and functions.
 --
+-- @usage
+--    -- Create a widget and update its content using the output of a shell
+--    -- command every 10 seconds:
+--    local mybatterybar = wibox.widget {
+--        {
+--            min_value    = 0,
+--            max_value    = 100,
+--            value        = 0,
+--            paddings     = 1,
+--            border_width = 1,
+--            forced_width = 50,
+--            border_color = "#0000ff",
+--            id           = "mypb",
+--            widget       = wibox.widget.progressbar,
+--        },
+--        {
+--            id           = "mytb",
+--            text         = "100%",
+--            widget       = wibox.widget.textbox,
+--        },
+--        layout      = wibox.layout.stack,
+--        set_battery = function(self, val)
+--            self.mytb.text, self.mypb.value = tonumber(val).."%", tonumber(val)
+--        end,
+--    }
+--
+--    gears.timer {
+--        timeout   = 10,
+--        autostart = true,
+--        callback  = function()
+--            -- You should read it from `/sys/class/power_supply/` (on Linux)
+--            -- instead of spawning a shell. This is only an example.
+--            awful.spawn.easy_async(
+--                {"sh", "-c", "acpi | sed -n 's/^.*, \([0-9]*\)%/\1/p'"},
+--                function(out)
+--                    mybatterybar.battery = out
+--                end
+--            )
+--        end
+--    }
+--
```

Comment:
Ok, I will add more magic to be able to add shim commands to `$PATH`. That way we will have a stable output.

> It's a nice example. :-)

(as you know) coming from a genuine question. He was kind of right, there was bits of answer everywhere, but putting them together was non trivial.

The signal/connect part of the code is still undocumented (there is no `client:connect_signal` documentation because I have yet to create a shared version for CAPI, the widgets have one, but without an example)